pkg/ingress/kube/common/tool.go

```diff
@@ -193,7 +196,29 @@ func SortHTTPRoutes(routes []*WrapperHTTPRoute) {
 		}
 
 		if routes[i].OriginPathType == routes[j].OriginPathType {
-			return len(routes[i].OriginPath) > len(routes[j].OriginPath)
+			if n, m := len(routes[i].OriginPath), len(routes[j].OriginPath); n != m {
+				return n > m
+			}
+
+			m1, m2 := routes[i].HTTPRoute.Match[0], routes[j].HTTPRoute.Match[0]
+			// methods
+			if n, m := len(m1.Method.GetRegex()), len(m2.Method.GetRegex()); true {
+				if n != 0 && m != 0 {
+					return n < m
```

Comment:
When n==m, we need check other conditions.

Reply:
ok